parser/prism/Translator.cc

```diff
@@ -119,9 +119,17 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
             auto blockParamNode = reinterpret_cast<pm_block_parameter_node *>(node);
             pm_location_t *loc = &blockParamNode->base.location;
 
-            std::string_view name = parser.resolveConstant(blockParamNode->name);
+            core::NameRef sorbetName;
+            if (auto prismName = blockParamNode->name; prismName != PM_CONSTANT_ID_UNSET) {
+                // A named block parameter, like `def foo(&block)`
+                auto name = parser.resolveConstant(prismName);
+                sorbetName = gs.enterNameUTF8(name);
+            } else { // An anonymous block parameter, like `def foo(&)`
+                sorbetName =
+                    gs.freshNameUnique(core::UniqueNameKind::Parser, core::Names::ampersand(), ++uniqueCounter);
+            }
```

Comment:
This all makes complete sense to me. Would it make sense to extract this into a `sorbetName` helper?

Reply:
There's a strange inconsistency with `parser::Restarg`, which needs the `nameLoc` to be modified conditionally :(